From 911f5113cbde1526109c1171f966c45eeced5eb8 Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Tue, 26 Dec 2023 10:46:43 +0000
Subject: [PATCH 1/1] Adjust links and wrapper scripts in bin/ for a headless
 build

- use 'soffice.bin' as a target binary instead of a 'soffice' shell script
  which is not installed in case of building without X support
- drop a useless wrapper script 'lofromtemplate'
---
 bin/distro-install-desktop-integration | 5 ++---
 sysui/desktop/share/create_tree.sh     | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/bin/distro-install-desktop-integration b/bin/distro-install-desktop-integration
index 7e1428ffba69..a3b6b4f6887f 100755
--- a/bin/distro-install-desktop-integration
+++ b/bin/distro-install-desktop-integration
@@ -152,9 +152,8 @@ add_wrapper lomath         soffice "--math"       "libreoffice" "libreoffice-mat
 add_wrapper loimpress      soffice "--impress"    "libreoffice" "libreoffice-impress.desktop"     "gid_Module_Brand_Prg_Impress"
 add_wrapper loweb          soffice "--web"        "libreoffice" ""                                "gid_Module_Brand_Prg_Wrt"
 add_wrapper lowriter       soffice "--writer"     "libreoffice" "libreoffice-writer.desktop"      "gid_Module_Brand_Prg_Wrt"
-add_wrapper lofromtemplate soffice ".uno:NewDoc"  "libreoffice" ""                                "gid_Module_Root_Brand"
 add_wrapper libreoffice    soffice ""            "libreoffice" "libreoffice-startcenter.desktop" "gid_Module_Root_Brand"
-add_wrapper loffice        soffice ""            "libreoffice" ""                                "gid_Module_Root_Brand"
+add_wrapper loffice        soffice.bin ""            "libreoffice" ""                                "gid_Module_Root_Brand"
 add_wrapper unopkg         unopkg  ""            "unopkg"      ""                                "gid_Module_Root_Brand"
 
 # there are two more desktop files for optional filters
@@ -167,7 +166,7 @@ test -f $DESTDIR/gid_Module_Optional_Xsltfiltersamples && echo "$PREFIXDIR/share
 #       you cannot include this link directly into the package
 #       For example, the Novell package mark this symlink as %ghost
 #	and update it in %post and %postun
-ln -sf "$INSTALLDIR/program/soffice" "$DESTDIR$BINDIR/soffice"
+ln -sf "$INSTALLDIR/program/soffice.bin" "$DESTDIR$BINDIR/soffice"
 test -f $DESTDIR/gid_Module_Root_Brand && echo "$BINDIR/soffice" >>$DESTDIR/gid_Module_Root_Brand
 
 # create bash completion
diff --git a/sysui/desktop/share/create_tree.sh b/sysui/desktop/share/create_tree.sh
index ce6475cd5ca5..111b850ed2d5 100755
--- a/sysui/desktop/share/create_tree.sh
+++ b/sysui/desktop/share/create_tree.sh
@@ -37,7 +37,7 @@ test -n "${OFFICE_PREFIX}" && office_prefix="${OFFICE_PREFIX}" || office_prefix=
 office_root=${office_prefix}/${PREFIX}
 
 #this symlink is needed to have the API bootstrap functions running right
-ln -sf "${office_root}/program/soffice" "${DESTDIR}/${BINDIR}/${PREFIX}"
+ln -sf "${office_root}/program/soffice.bin" "${DESTDIR}/${BINDIR}/${PREFIX}"
 
 if test "${PREFIX}" != libreoffice${PRODUCTVERSION} -a "${PREFIX}" != libreofficedev${PRODUCTVERSION}  ; then
     # compat symlinks
-- 
2.43.0

